src/services/gquery.js

```diff
@@ -77,9 +77,13 @@ class GQuery {
    * queries.
    */
 
-  constructor () {
+  constructor (enableWebSockets = false) {
+    const baseUrl = `${window.location.hostname}${window.location.port ? ':' + window.location.port : ''}${window.location.pathname}`
+    const httpUrl = `${window.location.protocol}//${baseUrl}graphql`
```

Comment:
Is `window.location.pathname` guaranteed to end with a trailing `/`?

Reply:
In which case is there a risk `httpUrl` could be invalid?

Reply:
Yes, I think there is no way to guarantee it will work 100% times. As it would depend on how the user deployed it, and it also relies on the browser to return the value with that trailing `/`.

I think we used to have a `TODO` marker for it somewhere when the code was first created but I probably removed it. I will add a `TODO` and an issue on GitHub where we can discuss and see what's the best way to specify the URL or at least the basepath.

Reply:
Done, both the `TODO`, and the [issue](https://github.com/cylc/cylc-ui/issues/343)